Thrive.csproj

```diff
@@ -21,6 +21,10 @@
   <ItemGroup>
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="simulation_parameters\CompoundLoader.cs" />
+    <Compile Include="src\auto-evo\simulation\niche\ChemosyntheticNiche.cs" />
+    <Compile Include="src\auto-evo\simulation\niche\INiche.cs" />
+    <Compile Include="src\auto-evo\simulation\niche\HeterotrophicNiche.cs" />
+    <Compile Include="src\auto-evo\simulation\niche\PhotosyntheticNiche.cs" />
```

Comment:
I and one other person (who didn't comment) here think that it is unnecessary or even harmful to have hardcoded niches. Niches should automatically form through some algorithm, so there shouldn't be multiple class types that are used to represent niches.

Reply:
Thanks for letting me know, but I won't be able to meet that requirement. I'll close the PR.

Reply:
Why not? Even if the rules are a bit arbitrary now, it could be a good thing to add the niche concept to the algorithm, and that should be pretty easy?

Reply:
I'm sure there's plenty of code you can use here, and you're free to do so, but taking ownership of a PR all the way to the finish line takes effort, as my other PR shows. I'm passionate about OO, and I want to make code I'm proud of. I'm not going to tell you how to run your codebase (this is the cleanest non-solo project I've seen), and working within these parameters isn't going to be a good time for me. Please take whatever you want from what I wrote, but somebody else needs to be responsible for linting and testing.

Reply:
I can understand if you don't have the time or patience to get a PR finished. 
But I want to clarify on my request about not having the separate classes being a bad design by itself, or that doing OOP is not fine in the Thrive codebase, instead it's about that in the past I've seen that when we say that "we are going to have these categories for x because that's how it is on Earth" the community reaction is that we are restricting things too much, as Thrive is not happening on Earth. So that's why the community probably thinks, and a few of us developers agree, that having hard coded niche categories on the code level, is not a good design. 
Instead the design can still be OO by having a Niche class, but one that is so flexible that it can represent any kind of niche. Though, we can discuss if that is also unsuitable design if handling wildly different niche archetypes makes a single class too unwieldy.

Reply:
Hey there adQuid, I like the concept, but part of contributing to an open source project is seeing your work through to completition to the satisfaction of the project maintainers, I have done liek what you did in teh past (i  just randomly implemented prokaryotic gameplay)  I like that passion but if you don't see prs through to completion their value drops dramatically. And also its kinda a pain to juggle one pr around different programmers.

Reply:
I have asked @Gwen-Benzschawel  if she would like to take on this rewrite, she is really passionate.  I like to read through new code but i don't actively contribute much to the project anymore. I see the value of fixing auto evo and implementing niches tho.